activerecord/test/cases/adapters/sqlite3/sqlite3_adapter_test.rb

```diff
@@ -345,6 +345,47 @@ def test_compound_index
         end
       end
 
+      def test_expression_index
+        skip unless @conn.supports_expression_index?
```

Comment:
Better than skipping the test is to only define if the connection supports it, like you did in the file below.